crypto/poseidon/codehash.go

```diff
@@ -0,0 +1,57 @@
+package poseidon
+
+import (
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto"
+)
+
+const defaultPoseidonChunk = 3
+
+func CodeHash(code []byte) (h common.Hash) {
+	// special case for nil hash
+	if len(code) == 0 {
+		return crypto.Keccak256Hash(nil)
```

Comment:
```suggestion
		return codehash.EmptyCodeHash
```

Reply:
I agree this would be better, but this leads to a cyclic dependency between `crypto/poseidon` and `crypto/codehash`.

Reply:
I see.